src/compute/src/render/reduce.rs

```diff
@@ -422,22 +435,26 @@ where
     fn build_distinct<S>(
         &self,
         collection: Collection<S, (Row, Row), Diff>,
-    ) -> (Arrangement<S, Row>, Collection<S, DataflowError, Diff>)
+    ) -> (
+        KeyValArrangement<S, Row, ()>,
+        Collection<S, DataflowError, Diff>,
+    )
     where
         S: Scope<Timestamp = G::Timestamp>,
     {
         let error_logger = self.error_logger();
 
         let (output, errors) = collection
-            .mz_arrange::<RowSpine<_, _, _, _>>("Arranged DistinctBy")
+            .map(|(k, _)| (k, ()))
```

Comment:
Please assert that `v` is empty, otherwise we'd be losing data.

Reply:
Done!